jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -708,7 +708,15 @@ def mos_niriss_parser(app, data_dir):
                         # The wavelength is stored in a WAVELENGTH HDU. This is
                         # a 2D array, but in order to be able to use Spectrum1D
                         # we use the average wavelength for all image rows
-                        wav = temp[wav_hdus[sci]].data.mean(axis=0) * u.micron
+
+                        if orientation == 'R':
```

Comment:
alternatively, should we not rely on this "orientation" and instead assume that the wavelength axis should be longer and change this check to `if data.shape[0] > data.shape[1]`?

Reply:
I think it is fair to assume that the dispersion axis is longer than the cross-dispersion axis and it will be simpler to adapt to other instruments that do not use the R/C nomenclature.

Reply:
ok, I switched to that logic.  But note that this is currently only implemented for the niriss parser (although we could copy similar logic elsewhere or consider a refactor to have this somewhere more general if needed down the road).  Thanks!

Reply:
Yes, but it will soon be time to extend to NIRCam ;)